### DISCUSSION TITLE
Custom error page `_error.js` crashes only in Vercel if getInitialProps is defined

### DISCUSSION THREAD

**Discussion by shivangg** [vercel/next.js#32043] [Help]
### What example does this report relate to?

with-sentry

### What version of Next.js are you using?

11.1.2, 12.0.4

### What version of Node.js are you using?

14.18

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

Vercel Serverless function crashes when error thrown in SSR. 

Also, this deviates from its behaviour when running locally with `yarn build && yarn start`.

### Expected Behavior

Should match the behaviour when run locally with `yarn build && yarn start`. 

Should render the custom error page and not crash.

### To Reproduce

Use the with-sentry example.

Also can be seen https://with-sentry-app.vercel.app/.

Test 1, Test 2, Test 3 - all of these crash the Vercel Serverless function.

![image](https://user-images.githubusercontent.com/10586637/142554426-b82c803a-ba10-40fe-8399-e11c1ddd1fc9.png)

**Comment by shivangg** [comment]
@timneutkens Can you confirm if this is a known issue with Vercel?

**Comment by ijjk** [comment]
Hi, this is currently the expected behavior due to how serverless functions need to be exited. To customize the error page `pages/500` should be used as documented here https://nextjs.org/docs/advanced-features/custom-error-page#500-page

**Comment by dan-turner** [comment]
@ijjk I don't understand, there is an example in the next.js repo and it doesn't work.  Are you able to please share more explicit instructions on how it should be changed to work?

https://github.com/vercel/next.js/blob/canary/examples/with-sentry

@shivangg were you able to figure it out?

**Comment by Entkenntnis** [comment]
We are facing the exact same problem: custom `_error` component is not rendered on vercel.

(but it's working locally as expected)

**Reply by MohammadPCh** [reply]
> We are facing the exact same problem: custom `_error` component is not rendered on vercel.
> 
> (but it's working locally as expected)

Can you find a solution?

**Reply by Entkenntnis** [reply]
Nope, we are using a static error page.

We are trying to do some customizing by checking for the path in the client

**Comment by clarkexcore** [comment]
We are experiencing this same issue as well. Serverless function errors are not being recorded in Sentry without a work around in the getServerSideProps. Has there been any solution for this yet?

**Comment by mitestainer** [comment]
Same here. We use a custom `_error.js` to record the errors on Sentry [as their docs suggest](https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#create-a-custom-_error-page). `_error.js` works on a local build but it simply doesn't work after deploying on Vercel - it shows that _Serverless Function crashed_ page. If we add a `500.js` page, then the code related to Sentry on `_error.js` doesn't run (and no errors are recorded, which is not correct).

Next v.12.3.1
Sentry v7.14.0
